master/internal/db/postgres_jobs_intg_test.go

```diff
@@ -0,0 +1,168 @@
+		assert.Equal(t, sendJob.OwnerID, recvJob.OwnerID)
+		assert.Equal(t, decimal.NewFromInt(10).Equal(recvJob.QPos), true)
+	})
+
+	t.Run("retrieve non-existent job", func(t *testing.T) {
+		// retrieve job and test for equality
+		recvJob, err := db.JobByID(model.NewJobID())
+		require.Error(t, err)
+		require.Nil(t, recvJob)
+	})
+}
+
+func TestUpdateJobPosition(t *testing.T) {
+	db := setupDBForTest(t)
+
+	t.Run("update position", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position
+		err = db.UpdateJobPosition(sendJob.JobID, newPos)
+		require.NoError(t, err)
+
+		// retrieve job and confirm pos update
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, newPos.Equal(recvJob.QPos), true)
+	})
+
+	t.Run("update position - negative value", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(-5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position
+		err = db.UpdateJobPosition(sendJob.JobID, newPos)
+		require.NoError(t, err)
+
+		// retrieve job and confirm pos update
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, newPos.Equal(recvJob.QPos), true)
+	})
+
+	t.Run("update position - empty ID", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position
+		err = db.UpdateJobPosition(model.JobID(""), newPos)
+		require.Error(t, err)
+
+		// retrieve job and ensure queue pos not updated
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, origPos.Equal(recvJob.QPos), true)
+	})
+
+	t.Run("update position - ID does not exist", func(t *testing.T) {
+		// create and send job
+		origPos := decimal.NewFromInt(10)
+		newPos := decimal.NewFromInt(5)
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    origPos,
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// update job position for a job that doesn't exist
+		err = db.UpdateJobPosition(model.NewJobID(), newPos)
+		require.NoError(t, err)
+	})
+}
+
+func setupDBForTest(t *testing.T) *PgDB {
+	require.NoError(t, etc.SetRootPath(RootFromDB))
+
+	db := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, db, MigrationsFromDB)
+	return db
+}
```

Comment:
nit: replace this with TestMain -- see other TestMains in the repo for more context. Then, you can reference it in the other test functions with db.SingleDB()
https://medium.com/goingogo/why-use-testmain-for-testing-in-go-dafb52b406bc

Reply:
Thanks!!

Reply:
`TestMain(...)` is so clean. Unfortunately, the `db` package has a lot of integration test files, some of which are closing db connections in an attempt to ensure they are using a fresh/clean database. Details here: https://hpe-aiatscale.atlassian.net/browse/RM-27

This means that until they are refactored to their own package, tests need to establish their own database connection instead of using `TestMain(...)`.

Reply:
It was a good experience to learn about `TestMain` and implement the change even if it didn't work out! Now I know to be aware of it in the future. :)